go/teams/teams.go

```diff
@@ -1672,12 +1677,36 @@ func (t *Team) PostTeamSettings(ctx context.Context, settings keybase1.TeamSetti
 		Public:   t.IsPublic(),
 	}
 
-	err = t.postChangeItem(ctx, section, libkb.LinkTypeSettings, nil, sigPayloadArgs{})
+	payloadArgs := sigPayloadArgs{}
+	var maybeEKPayload *teamEKPayload
+	if rotate {
+		// Create empty Members section. We are not changing memberships, but
+		// it's needed for key rotation.
+		memSet := newMemberSet()
+		section.Members, err = memSet.Section()
+		if err != nil {
+			return err
+		}
+		secretBoxes, perTeamKeySection, teamEKPayload, err := t.rotateBoxes(ctx, memSet)
+		if err != nil {
+			return err
+		}
+		section.PerTeamKey = perTeamKeySection
+		payloadArgs.secretBoxes = secretBoxes
+		payloadArgs.teamEKPayload = teamEKPayload
+		maybeEKPayload = teamEKPayload // for storeTeamEKPayload, after post succeeds
+	}
+	err = t.postChangeItem(ctx, section, libkb.LinkTypeSettings, nil, payloadArgs)
 	if err != nil {
 		return err
 	}
 
-	t.notify(ctx, keybase1.TeamChangeSet{Misc: true})
+	if rotate {
+		t.notify(ctx, keybase1.TeamChangeSet{KeyRotated: true, Misc: true})
```

Comment:
this needs to change since i fixed sequence number accounting. That's likely what the merge conflict is...